modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-shipment-test/src/testIntegration/java/com/liferay/headless/commerce/admin/shipment/resource/v1_0/test/ShipmentItemResourceTest.java

```diff
@@ -106,6 +106,16 @@ public void setUp() throws Exception {
 				_commerceOrder.getShippingAddressId(),
 				_commerceOrder.getCommerceShippingMethodId(),
 				_commerceOrder.getShippingOptionName(), _serviceContext);
+
+		_newCommerceInventoryWarehouse = _getNewCommerceInventoryWarehouse();
```

Comment:
why do we need a new one? I mean... in the randomShipmentItem we use the ERC of the already created item like we do for the id.
So testing for the ERC should be: using 0 as id (but its ERC) is it correct that as response the warehouseId is equal to the id of "where I took the ERC"?

Reply:
Probably you need a new one for the patch... but in this case it's better to create it only in the specific tests

Reply:
In case is needed let's call this _addCommerceInventoryWarehouse and the variable _commerceInventoryWarehouse2 (and the already existing _commerceInventoryWarehouse1)

Reply:
You are right, this is required only for the put if I remember correctly, I will try to make it so that it is used only for that one.

Reply:
Here _addCommerceInventoryWarehouse().	getCommerceInventoryWarehouseId() is required only for the put.

Reply:
The one in line 110 is required for both put and patch. We could move this WH creation to the relevant private methods: _testPatchShipmentItemWithWarehouseExternalReferenceCode / _testPutShipmentByExternalReferenceCodeItemWithWarehouseExternalReferenceCode
That was my first approach, but this way we can spare some lines of code and the test is not getting slower, perhaps just a tiny bit.
Would you advise to put it back to the relevant methods?

Reply:
What's your opinion about line 340-341? That's only required for the _testPutShipmentByExternalReferenceCodeItemWithWarehouseExternalReferenceCode if I remember correctly. Should I "fix" this too? The old code was to simply get the already created WH's ERC.

Reply:
if needed is fine to have it here, no problem... just rename it as for the other comment. Thx